gpflow/kernels/linears.py

```diff
@@ -1,31 +1,36 @@
 import tensorflow as tf
-from ..base import Parameter, positive
+
 from .base import Kernel
+from ..base import Parameter, positive
 
 
 class Linear(Kernel):
     """
-    The linear kernel.  Functions drawn from a GP with this kernel are linear, i.e. f(x) = cx.
+    The linear kernel. Functions drawn from a GP with this kernel are linear, i.e. f(x) = cx.
     The kernel equation is
 
         k(x, y) = σ²xy
 
-    where σ²  is the variance parameter.
+    where σ² is the variance parameter.
     """
 
-    def __init__(self, variance=1.0, active_dims=None, ard=None):
+    def __init__(self, variance=1.0, active_dims=None):
         """
-        - input_dim is the dimension of the input to the kernel
-        - variance is the (initial) value for the variance parameter(s)
-          if ard=True, there is one variance per input
-        - active_dims is a list of length input_dim which controls
-          which columns of X are used.
+        :param variance: the (initial) value for the variance parameter(s),
+            to induce ARD behaviour this must be initialised as an array the same
+            length as the the number of active dimensions e.g. [1., 1., 1.]
+        :param active_dims: a slice or list specifying which columns of X are used
         """
         super().__init__(active_dims)
-
-        # variance, self.ard = self._validate_ard_shape("variance", variance, ard)
-        self.ard = ard
         self.variance = Parameter(variance, transform=positive())
+        self._validate_ard_active_dims(self.variance)
+
+    @property
+    def ard(self) -> bool:
+        """
+        Whether ARD behaviour is active.
+        """
+        return len(self.variance.shape) > 0
```

Comment:
self.variance.shape.ndims
```suggestion
        return self.variance.shape.ndims > 0
```